src/averageAges.js

```diff
@@ -20,6 +20,30 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  let num = 0;
+  const sum = people.reduce(calculateSum, 0);
+
+  function calculateSum(prev, person) {
+    if (century) {
+      if (person.sex === 'm' && Math.ceil(person.died / 100) === century) {
+        num++;
+
+        return prev + (person.died - person.born);
+      }
+    }
+
+    if (!century) {
+      if (person.sex === 'm') {
+        num++;
+
+        return prev + (person.died - person.born);
+      }
+    }
```

Comment:
using nested conditions is a bad approach, consider using ternary or something like that